File: taller/AMarotoChivite-taller/AMarotoChivite-taller-Java/src/main/java/simulacionTaller/factories/BossFactory.java
package simulacionTaller.factories;

import simulacionTaller.models.Boss;
import simulacionTaller.models.Worker;

import static simulacionTaller.utils.randomInfo.*;

public class BossFactory {

    public static Boss create() {
        String name = randomName();
        int experienceYear = randomAnno();
        if (experienceYear < 5) {
            experienceYear = experienceYear + 5;
        }
        int salary = randomSalary() + 750;
        // Por defecto 0, ya que posteriormente cuando se cree todo el taller, se detallará a cada jefe sus trabajadores
        Worker[] workers = new Worker[0];
        return new Boss(name, experienceYear, salary, workers);
    }
}

File: taller/AMarotoChivite-taller/AMarotoChivite-taller-kotlin/src/main/kotlin/simulacionTaller/factories/BossFactory.kt
package simulacionTaller.factories

import simulacionTaller.models.Boss
import simulacionTaller.models.Worker
import simulacionTaller.utils.randomAnno
import simulacionTaller.utils.randomName
import simulacionTaller.utils.randomSalary

object BossFactory {
    fun create(): Boss {
        val name = randomName()
        var experienceYear = randomAnno()
        if (experienceYear < 5) {
            experienceYear += 5
        }
        val salary = randomSalary() + 750
        // Por defecto 0, ya que posteriormente cuando se cree todo el taller, se detallará a cada jefe sus trabajadores
        val workers = arrayOfNulls<Worker>(0)
        return Boss(name, experienceYear, salary.toDouble(), workers)
    }
}